src/DependencyInjection/DI/test/ServiceProviderValidationTests.cs

```diff
@@ -109,6 +109,30 @@ public void GetService_DoesNotThrow_WhenScopeFactoryIsInjectedIntoSingleton()
             Assert.NotNull(result);
         }
 
+        [Fact]
+        public void BuildServiceProvider_ThrowsForUnresolvableServices()
```

Comment:
Can you add some more interesting cases? IEnumerable, open generic etc.

Reply:
Open generics?

Reply:
So you loop over the service collection and call GetService. What happens when you encounter IFoo<>, Foo<> for example. What ends up happening?

Reply:
Whoops.

Reply:
Done